src/Form/QueueContentsForm.php

```diff
@@ -0,0 +1,173 @@
+   * @throws \Drupal\media_mpx\Exception\MediaTypeNotAssociatedWithMpxException
+   */
+  public function batchedQueueItemsForMediaType(string $media_type_id, array &$context) {
+    /* @var \Drupal\media_mpx\Service\QueueVideoImports $service */
+    $service = \Drupal::getContainer()->get('media_mpx.service.queue_video_imports');
+    $batch_size = 200;
+
+    if (empty($context['sandbox'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['last_index'] = 0;
+      $context['results'] = [
+        'success' => 0,
+        'errors' => 0,
+      ];
+    }
+
+    $request = new QueueVideoImportsRequest($media_type_id,
+      $batch_size,
+      $context['sandbox']['last_index'] + 1);
+
+    $response = $service->execute($request);
+
+    if (!isset($context['sandbox']['total_results'])) {
+      $context['sandbox']['total_results'] = $response->getIterator()->getList()->getTotalResults();
+    }
+    $total_items = $context['sandbox']['total_results'];
+
+    $context['results']['success'] += $response->getVideosQueued();
+    $context['results']['errors'] += $response->getErrors();
+    $context['sandbox']['progress'] += $response->getVideosQueued();
+    $context['sandbox']['last_index'] = $context['sandbox']['last_index'] + $batch_size;
+
+    $context['message'] = 'Processed a total of ' . $context['sandbox']['progress'] . ' items.';
+    $context['message'] = t('Queued @progress %type items out of @total.', [
+      '@progress' => $context['sandbox']['progress'],
+      '@total' => $total_items,
+      '%type' => $media_type_id,
+    ]);
+
+    if ($context['sandbox']['progress'] != $total_items) {
+      $context['finished'] = $context['sandbox']['progress'] / $total_items;
+    }
+  }
+
+  /**
+   * Finished callback for the batch queuing operations to update mpx videos.
+   *
+   * @param bool $success
+   *   Result of the queuing operations.
+   * @param array $results
+   *   An array with the count of successful and errored queue operations.
+   */
+  public function batchFinished(bool $success, array $results) {
+    $finished_message = t('Finished with errors.');
+    $messenger = \Drupal::messenger();
+
+    if ($success) {
+      $finished_message = \Drupal::translation()->formatPlural(
+        $results['success'],
+        'One mpx item queue.',
+        '@count mpx items queued for updates.');
+    }
+
+    $messenger->addMessage($finished_message);
+
+    if ($results['errors'] > 0) {
+      $errors_message = \Drupal::translation()->formatPlural(
+        $results['errors'],
+        'One mpx item could not be queued (check logs for details).',
+        '@count mpx items could not be queued for updates (check logs for details).');
+      $messenger->addError($errors_message);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'media_mpx_asset_sync_queue';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
```

Comment:
A good convention is to place create right after the constructor. That way, if arguments change this method is easy to see that it exists.